lib/adapty_flutter.dart

```diff
@@ -198,6 +199,17 @@ class Adapty {
     return result ?? false;
   }
 
+  static Future<MakePurchaseResult?> changeSubscription(AdaptyProduct newProduct, AdaptyAndroidSubscriptionUpdateParams subscriptionUpdateParams) async {
```

Comment:
Is there a specific reason why this should be a separate function from `makePurchase`? i.e. you could potentially reuse the makePurchase function with optional named parameters, that way developers don't necessarily need to take into account users switching products, and if they do, they can provide the optional named param

 e.g.

```dart
static Future<MakePurchaseResult> makePurchase(AdaptyProduct product, { AdaptyAndroidSubscriptionUpdateParams? androidSubscriptionUpdateParams }) async {
    final result = (await _invokeMethodHandlingErrors<String>(Method.makePurchase, {
      Argument.productId: product.vendorProductId,
      if (product.variationId != null) Argument.variationId: product.variationId,
      if (androidSubscriptionUpdateParams != null) Argument.params: androidSubscriptionUpdateParams.toMap()
    })) as String;
    return MakePurchaseResult.fromJson(json.decode(result));
  }
```

Reply:
@danieljimeneznz thank you for the comment, @x401om @NiyazFazlyev your opinion?

Reply:
@vladd-g Thinking about this some more, the main gripe I have with this method is the `if (!Platform.isAndroid) return null;` because this function should be platform agnostic. 

From a usability standpoint this becomes quite laborious because when I come to use the `changeSubscription()` method I have to wrap my code in another `Platform.isAndroid` call i.e.

```dart
void changeSubscription() {
  if (Platfrom.isAndroid) Adapty.changeSubscription();
  if (Platfrom.isIOS) Adapty.makePurchase();
}
```

Where-as I'd much prefer to call the function like below, and have the library handle the iOS/Android specific function calls
```dart
void changeSubscription() => Adapty.changeSubscription();
```

Reply:
@danieljimeneznz Unfortunately, you may need to do platform check anyway since you need to provide `AdaptyAndroidSubscriptionUpdateParams` on Android because Play Market doesn't know what subscription you want to change and under what conditions. Unlike AppStore where there are subscription groups out of the box, and thus the flow for both buying a new sub and changing existing is the same on the client side afaik.

Reply:
Good point! What about something like:

My Mobile application code:
```dart
Future<void> changeSubscription(product) async {
  AdaptyAndroidSubscriptionUpdateParams? params = Platform.isAndroid ? AdaptyAndroidSubscriptionUpdateParams() : null;
  await Adapty.changeSubscription(product, androidSubscriptionUpdateParams: params);
}
```

Library code:
```dart
static Future<MakePurchaseResult> makePurchase(AdaptyProduct product, { AdaptyAndroidSubscriptionUpdateParams? androidSubscriptionUpdateParams }) async {
    final result = (await _invokeMethodHandlingErrors<String>(Method.makePurchase, {
      Argument.productId: product.vendorProductId,
      if (product.variationId != null) Argument.variationId: product.variationId,
      if (androidSubscriptionUpdateParams != null) Argument.params: androidSubscriptionUpdateParams.toMap()
    })) as String;
    return MakePurchaseResult.fromJson(json.decode(result));
  }
  
static Future<MakePurchaseResult> changeSubscription(AdaptyProduct newProduct, { AdaptyAndroidSubscriptionUpdateParams? androidSubscriptionUpdateParams }) async =>
  makePurchase(newProduce, androidSubscriptionUpdateParams: androidSubscriptionUpdateParams);
```

This has a couple benefits, in my code, regardless of platform I am always calling `changeSubscription` to change the subscription (no `makePurchase` call), the `changeSubscription` is a separate function in the library so should the logic need to be changed it can be easily modified, and it is utilising the already available `makePurchase` code so there is less duplication of the invoked method.

Alternatively if no androidSubscriptionUpdate params are provided to the `changeSubscription` function the default proration mode could be injected by the library i.e. https://developer.android.com/google/play/billing/subscriptions#change (IMMEDIATE_WITH_TIME_PRORATION) is the default Google Play subscription change proration mode.

Reply:
(I'm taking an interest in this PR as writing upgrade/downgrade subscription logic is the next thing in my app's to-do list 😄)

Reply:
@danieljimeneznz If no androidSubscriptionUpdate params are provided, it would be interpreted as buying yet another subscription, since besides proration mode you also need to provide a `vendorProductId` of the subscription to be replaced